survey_formio/models/survey.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2023 Acsone SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import json
+from datetime import datetime, timedelta
+
+from odoo import models
+
+
+class SurveySurvey(models.Model):
+    _inherit = "survey.survey"
+
+    def _get_formio_display(self):
+        """
+        Returns a string representing the form.io display
+        Correct values are form, wizard or pdf. At the moment only
+        form is supported.
+        In the future use questions_layout to determine display.
+        """
+        self.ensure_one()
+        return "form"
+
+    def _get_components(self, day_first=True):
+        """
+        Pages are not supported yet.
+
+        Returns a list of dict representing survey questions as form.io components
+
+        Parameters:
+        day_first: boolean that determines if the day comes first or
+                   not in dates
+        """
+        self.ensure_one()
+        components = []
+        for question in self.question_ids:
+            components.append(question._get_formio_dict(day_first))
+        return components
+
+    def generate_formio_json(self, day_first=True):
+        """
+        Returns a form.io conform json representation of a survey.
+
+        Parameters:
+        day_first: boolean that determines if the day comes first or
+                   not in dates
+        """
+        self.ensure_one()
+        display = self._get_formio_display()
+        title = self.title
+        components = self._get_components(day_first)
+
+        form_json = json.dumps(
+            {"display": display, "title": title, "components": components}
+        )
+        return form_json
+
+    def user_input_from_formio(self, formio_input, user_input_id=None, day_first=True):
```

Comment:
This function is a big long. Can it be split in small functions for each answer type?